### PR TITLE
Move SourceHandler to the new select syntax

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR moves the SourceHandler to the new select syntax.

### CODE REVIEW

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -669,18 +662,16 @@ def get_sources(
                 )
             )
         )
-        classification_accessible_subquery = Classification.query_records_accessible_by(
-            user
-        ).subquery()
+        classification_accessible_subquery = Classification.select(user).subquery()
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:665]: I'm not sure if this has been defined twice or why?

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -740,6 +731,14 @@ def get_sources(
                 "Invalid commentsFilterAuthor value -- must provide at least one string value"
             )
     if localization_dateobs is not None:
+
+        # This grabs just the IDs so the more expensive localization in-out
+        # check is done on only this subset
+        obj_ids = session.scalars(obj_query).all()
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:737]: What happens if the filters above don't filter anything and you get back *all* the obj ids?

File: skyportal/handlers/api/source.py [hunk 3]
```diff
@@ -740,6 +731,14 @@ def get_sources(
...
+        obj_query = Obj.select(user, options=obj_query_options).where(
+            Obj.id.in_(obj_ids)
+        )
+
         if localization_name is not None:
             localization = (
                 Localization.query_records_accessible_by(user)
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:744]: You forgot one `query_records_accessible_by`

File: skyportal/handlers/api/source.py [hunk 4]
```diff
@@ -1203,7 +1203,11 @@ def post_source(data, user_id, session):
 
     user = session.query(User).get(user_id)
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:1204]: This can also be updated to new style.

File: skyportal/handlers/api/source.py [hunk 5]
```diff
@@ -1232,20 +1236,20 @@ def post_source(data, user_id, session):
...
+        session.add(obj)
 
     if (ra is not None) and (dec is not None):
         # This adds a healpix index for a new object being created
         obj.healpix = ha.constants.HPX.lonlat_to_healpix(ra * u.deg, dec * u.deg)
 
-    groups = (
-        Group.query_records_accessible_by(user).filter(Group.id.in_(group_ids)).all()
-    )
+    groups = session.scalars(Group.select(user).where(Group.id.in_(group_ids))).all()
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:1252]: Should we also test that all groups requested by `group_ids` are accessible (i.e., they were indeed loaded by the select)?

File: skyportal/handlers/api/source.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:665]: There is one for classifications and one for nonclassifications.

File: skyportal/handlers/api/source.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:737]: I will put a cap and raise an error.

File: skyportal/handlers/api/source.py (hunk 4 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:1204]: good idea.

File: skyportal/handlers/api/source.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:744]: oops.

File: skyportal/handlers/api/source.py (hunk 5 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:1252]: good idea.

File: skyportal/handlers/api/source.py [hunk 6]
```diff
@@ -173,26 +168,30 @@ def get_source(
...
+            session.scalars(
+                Comment.select(
+                    user,
+                    options=[
+                        joinedload(Comment.author),
+                        joinedload(Comment.groups),
+                    ],
+                )
+                .where(Comment.obj_id == obj_id)
+                .distinct()
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/source.py]: Removed distinct.

**Review by guynir42 — APPROVED** [review]

**Pull request merged**